src/libraries/Microsoft.PowerFx.Core/Public/Values/CollectionTableValue.cs

```diff
@@ -85,5 +87,17 @@ record = Marshal(item);
                 return base.TryGetIndex(index1, out record);
             }
         }
+
+        public bool Append(T record)
```

Comment:
>bool [](http://example.com/codeflow?start=15&length=4)

This should be a virtual on TableValue (like Count, TryGetIndex). 
The default implementation can error. (for now, throwing is ok - we'll confirm the right error value later). 

Then this becomes:
            if (_sourceList == null)
            {
                return base.Append(record);
            }
            _sourceList.Add(record);

Reply:
Shouldn't this API be named TryAppend(...) instead?
If it's Append, it shouldn't return a bool, but void.

Reply:
+1 - if we do end up with bool, it should be TryAppend. 
Some of the PowerFx functions (like Set) always just return 'true'; but even in that case, the C# signature can be void, and the power fx function impl can handle the return. 

Although, sorting out exact semantics with Greg - leaning towards returning a FormulaValue for the newly added record (with any server-side fields filled out); or an error value (say the server rejected the add)

Reply:
Agreed. I'm leaving as it currently is since there is still a discussion about what the function should return.